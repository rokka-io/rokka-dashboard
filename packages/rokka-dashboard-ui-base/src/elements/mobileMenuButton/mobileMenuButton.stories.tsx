import { action } from '@storybook/addon-actions';
import { boolean } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import { colors } from '../../identity/colors/colors';
import { MobileMenuButton } from './mobileMenuButton';

storiesOf('Elements / Header', module)
  .addParameters({
    info: 'The MobileMenuButton is not visible on the desktop viewport, please resize storybook accordingly.'
  })
  .addDecorator(story => <div style={{ background: colors.brand.dark }}>{story()}</div>)
  .add('MobileMenuButton', () => <MobileMenuButton onClick={action('onClick')} active={boolean('active', false)} />);
