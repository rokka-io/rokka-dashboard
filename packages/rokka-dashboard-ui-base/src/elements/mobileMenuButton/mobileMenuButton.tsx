import React from 'react';
import styled from 'styled-components';
import { colors } from '../../identity/colors/colors';
import { media } from '../../identity/media/media';
import { spaces } from '../../identity/spaces/spaces';
import { fonts } from '../../identity/typography';

interface MobileMenuButtonProps {
  /** Whether button is active */
  active?: boolean;
  /** Callback */
  onClick?(): void;
}

export function MobileMenuButton({ onClick, active = false }: MobileMenuButtonProps) {
  return <StyledMobileMenuButton active={active} type="button" onClick={onClick} />;
}

interface StyledMobileMenuButtonProps {
  active?: boolean;
}

// TODO: move to own element
const LinkButton = styled.button`
  background-color: transparent;
  border: none;
  cursor: pointer;
  font-family: ${fonts.Families.brand};
  font-size: ${fonts.Sizes.small};
  line-height: inherit;
  display: inline;
  margin: 0;
  padding: 0;
`;

const StyledMobileMenuButton = styled(LinkButton)<StyledMobileMenuButtonProps>`
  transition: background 0.4s ease;
  display: inline-block;
  height: ${spaces.large};
  width: ${spaces.large};
  border-radius: 4px;
  color: ${colors.tints.white};
  font-size: ${fonts.Sizes.large};
  margin-right: ${spaces.medium};
  vertical-align: middle;
  position: relative;

  ${media.desktop`
    display: none;
  `}

  ${({ active }) =>
    active &&
    `
      border: 1px solid ${colors.tints.white};
      background: rgba(0,0,0,.1);
    `}

  &:after {
    top: 50%;
    left: 50%;
    position: absolute;
    content: 'â˜°';
    transform: translate(-50%, -50%);
  }
`;
