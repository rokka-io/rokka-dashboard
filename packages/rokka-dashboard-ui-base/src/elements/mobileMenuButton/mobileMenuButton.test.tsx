import { shallow } from 'enzyme';
import React from 'react';
import { MobileMenuButton } from './mobileMenuButton';

describe('Elements', () => {
  describe('Header', () => {
    it('renders a MobileMenuButton correctly.', () => {
      const spy = jest.fn();
      const component = shallow(<MobileMenuButton onClick={spy} />);

      component.simulate('click');

      expect(component).toMatchSnapshot();
      expect(spy).toHaveBeenCalledTimes(1);
    });
    it('renders an active MobileMenuButton correctly.', () => {
      const spy = jest.fn();
      const component = shallow(<MobileMenuButton active={true} onClick={spy} />);

      component.simulate('click');

      expect(component).toMatchSnapshot();
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });
});
