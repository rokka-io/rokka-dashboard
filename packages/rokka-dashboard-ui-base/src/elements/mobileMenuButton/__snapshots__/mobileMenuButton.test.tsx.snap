// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements Header renders a MobileMenuButton correctly. 1`] = `
<Styled(styled.button)
  active={false}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  type="button"
/>
`;

exports[`Elements Header renders an active MobileMenuButton correctly. 1`] = `
<Styled(styled.button)
  active={true}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  type="button"
/>
`;
